src/soc/ibm/power9/xbus.h

```diff
@@ -0,0 +1,75 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+
+#ifndef __SOC_IBM_POWER9_XBUS_H
+#define __SOC_IBM_POWER9_XBUS_H
+
+#include <stdint.h>
+
+/*
+ * Define DEBUG_XBUS before including this header to get debug prints from this
+ * unit
+ */
+
+#define XBUS_LANE_COUNT 17
+
+#define XBUS_LINK_GROUP_OFFSET 0x2000000000
+
+/*
+ * The API below is meant to be used after SBE for the second CPU is up (so
```

Comment:
Ideally there should be one, common API for all access methods that would be changed to the next method on command, without having to use different functions at highest level. For the sake of debugging lets leave it as it is for now.

Is my understanding below correct?
- first CPU always uses XSCOM
- second CPU <= 8.4: FSI (do we even use it for SCOM?)
- second CPU 8.5 (or whatever is after 8.4) - 10.1: SBEIO
- second CPU >10.1: XSCOM

Reply:
> FSI (do we even use it for SCOM?)

No, the code isn't in any of the branches (I have it locally somewhere).

Initially FSI SCOM isn't usable and second CPU is set up using FSI/CFAM. Then SBE is up and its SCOM can be used. FSI SCOM works only with 31-bit addresses and we need more than that, at the end it's kind of useless and I think Hostboot never uses it either.

> Is my understanding below correct?

Yes.